### PR TITLE
Support importing IPFS files

### PR DESCRIPTION
**Pull request by axic**
Implements #133. Blocked by not having CORS.

Since I've added this to try out, it may make sense having it here hoping it will be supported soon.

### CODE REVIEW

**Comment by axic** [comment]
There's still no CORS on the endpoint...

**Comment by axic** [comment]
@yann300 @chriseth this is working now 😉

**Comment by yann300** [comment]
cool!
If you have time could you just quick update the doc?
https://github.com/ethereum/remix/blob/master/docs/tutorial_import.rst
is this possible to push to ipfs also?

**Comment by axic** [comment]
> is this possible to push to ipfs also?

The public gateway only supports reading.

File: src/app.js [hunk 1]
```diff
@@ -557,6 +572,18 @@ var run = function () {
         files.addReadOnly(url, content)
         cb(null, content)
       })
+    } else if ((match = /^(ipfs:\/\/?.+)/.exec(url))) {
+      $('#output').append($('<div/>').append($('<pre/>').text('Loading ' + url + ' ...')))
+      handleIPFS(match[1], function (err, content) {
+        if (err) {
+          cb('Unable to import "' + url + '": ' + err)
```
Comment (chriseth) [inline src/app.js]: There is lots of code duplication between the three remote imports. Can you reduce this to something like
```js
var importers = [{match:  /^(https?:\/\/)?(www.)?github.com\/([^/]*\/[^/]*)\/(.*)/, importer = handleGithubImport}, {...}]
for (...) { ...}
```
?

Comment (axic) [inline src/app.js]: A planned to address that in a future PR. The only thing I've did in the direction is to move the log 😉

Comment (chriseth) [inline src/app.js]: Please to it in this pr, if possible. It's not a complicated change and three times the same code is already two times too much :-)

**Comment by axic** [comment]
@chriseth @yann300 updated to reduce code duplication.

**Comment by axic** [comment]
Moved the refactoring to a separate PR because it makes it really hard to review.

**Pull request merged**